src/platform/user/profile/utilities/index.js

```diff
@@ -123,22 +124,60 @@ export function mapRawUserDataToState(json) {
 export const hasSession = () => localStorage.getItem('hasSession');
 
 function compareLoginPolicy(loginPolicy) {
-  let attemptedLoginPolicy = localStorage.getItem(
-    authnSettings.PENDING_LOGIN_TYPE,
-  );
+  // This is experimental code related to GA that will let us determine
+  // if the backend is returning an accurate loginPolicy (service_name)
+
+  let attemptedLoginPolicy = localStorage.getItem('pendingLoginPolicy');
 
   attemptedLoginPolicy =
     attemptedLoginPolicy === 'mhv' ? 'myhealthevet' : attemptedLoginPolicy;
 
+  if (attemptedLoginPolicy === null) {
+    Raven.captureMessage(
+      "localStorage error: 'pendingLoginPolicy' was not stored",
+    );
+  }
+
   if (loginPolicy !== attemptedLoginPolicy) {
     recordEvent({
       event: `login-mismatch-${attemptedLoginPolicy}-${loginPolicy}`,
     });
   }
+
+  localStorage.removeItem('pendingLoginPolicy');
+}
+
+function recordGAAuthEvent(loginPolicy, loa) {
+  // The payload we receive from authenticating does not specify whether
+  // the user has logged in or if they are a new user. To differentiate, we are
+  // retrieving a "pendingAuthAction" stored in localStorage when the user
+  // clicks a sign in/register button in the Sign In Modal
+
+  const pendingAuthAction = localStorage.getItem('pendingAuthAction');
+
+  if (pendingAuthAction === 'register') {
+    // Record GA success event for the register method.
+    recordEvent({ event: `register-success-${loginPolicy}` });
+  } else if (pendingAuthAction === 'login') {
+    compareLoginPolicy(loginPolicy);
+    // Report GA success event for the login method.
+    recordEvent({ event: `login-success-${loginPolicy}` });
+  } else {
+    recordEvent({ event: `login-or-register-success-${loginPolicy}` });
+    Raven.captureMessage(
+      "localStorage error: 'pendingAuthAction' was not stored",
+    );
+  }
+
+  localStorage.removeItem('pendingAuthAction');
+
+  // Report out the current level of assurance for the user.
+  if (loa && loa.current) {
+    recordEvent({ event: `login-loa-current-${loa.current}` });
```

Comment:
I was originally thinking this could be called even with an active session, since my concern was that a `/user` request would result in a new user. And accordingly, we would report the LOA for that user as well. But it might be too much of an edge case to worry about.